std/string.d

```diff
@@ -5798,36 +5798,58 @@ unittest
  * function, or any of the conversion functions.
  *
  * Params:
- *     s = the string to check
+ *     s = the string or random access range to check
  *     bAllowSep = accept separator characters or not
  *
  * Returns:
  *     $(D bool)
  */
-bool isNumeric(S)(S s, bool bAllowSep = false) if (isSomeString!S)
+bool isNumeric(S)(S s, bool bAllowSep = false) if (isSomeString!S ||
+    (isRandomAccessRange!S &&
+    hasSlicing!S &&
+    isSomeChar!(ElementType!S) &&
+    !isInfinite!S))
 {
     import std.algorithm.comparison : among;
+    import std.ascii : isASCII;
+    import std.utf : byChar;
+
+    // auto-decoding special case, we're only comparing characters
+    // in the ASCII range so there's no reason to decode
+    static if (isSomeString!S)
+    {
+        import std.utf : byCodeUnit;
+        auto codeUnits = s.byCodeUnit;
+    }
+    else
+    {
+        alias codeUnits = s;
+    }
 
-    if (s.empty)
+    if (codeUnits.empty)
         return false;
 
     // Check for NaN (Not a Number) and for Infinity
-    if (s.among!((a, b) => icmp(a, b) == 0)
+    if (codeUnits.save.among!((a, b) => asciiCmp(a, b.byChar))
```

Comment:
Need to `.save` when calling `asciiCmp`. It's called multiple times on the first argument to `among`. Don't need to `.save` upfront then. I hadn't made this clear before, because I would have written it this way at first, too.

Test case that currently fails:
```d
dstring nani = "nani";
assert(isNumeric(refRange(&nani)));
```

Fixed code:
```d
if (codeUnits.among!((a, b) => asciiCmp(a.save, b.byChar))
```